strapi/strapi
# [plugin:documentation] add component schemas

### What does it do?

- Adds back [schema components](https://swagger.io/docs/specification/components/)
- Creates Response, ListResponse, and Request components 

### Why is it needed?

Components were not included in the v4 migration. 

### How to test it?

Run the `examples/getstarted` app then go to the documentation plugin and click `open documentation`. On the swagger UI you should see all the endpoints and a dropdown at the bottom for `schemas`. 

- Look at all the schemas and verify the response/request formats are correct (Some won't be, such as address, since the overrides still don't work).
- Look at all the endpoints and verify the correct schema is being used
- Try to 'GET', 'POST', 'PUT', 'DELETE' on different endpoints and confirm the response/error is correct

You could also run the plugin on Strapi v3.6.8 for comparison since this PR tries to implement what existed before

### Related issue(s)/PR(s)

resolves: https://github.com/strapi/strapi/issues/12297
resolves: https://github.com/strapi/strapi/issues/6088
